api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -106,7 +107,11 @@ message HttpConnectionManager {
     // A list of header names used to create tags for the active span. The header name is used to
     // populate the tag name, and the header value is used to populate the tag value. The tag is
     // created if the specified header name is present in the request's headers.
-    repeated string request_headers_for_tags = 2;
+    //
+    // .. attention::
+    //  This field has been deprecated in favor of :ref:`custom_tags
+    //  <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.tracing.custom_tags>`.
+    repeated string request_headers_for_tags = 2 [deprecated = true];
```

Comment:
Do you have a deprecated config test for this somewhere? I would expect some existing test to fail because you are using deprecated config (DEPRECATED_FEATURE_TEST)?

Reply:
here, i kept the original configuration item, and validated it in the new mode.
https://github.com/envoyproxy/envoy/blob/015dd4d0378bccfe566a8e53202a9dfe81d4821b/test/extensions/filters/network/http_connection_manager/config_test.cc#L158

Reply:
OK got it, thanks. In config_test.cc can you change `parseHttpConnectionManagerFromV2Yaml` to use `loadFromYamlAndValidate`? This will cause that test to fail with a deprecated feature use test in the compile time options build. Then, please split out the deprecated option into its own test (using DEPRECATED_FEATURE_TEST) so that it can be easily deleted later. Thank you!

Reply:
I changed and split it.

Not quite sure what i missed, when i only change it to use validate method without remove any deprecated case, it also passed. Seems that deprecated validation not work, i checked the generated validate.cc and no deprecated validation found.

Reply:
I've seen the failed case,I'll check it.

Reply:
fixed